src/main/steps/tasklistresponse/proceedings/navigationController.ts

```diff
@@ -103,6 +82,49 @@ class OtherProceedingsNavigationController {
     }
     return nextUrl;
   }
-}
 
+  private getNextUrlOtherProceedingDetails(caseData) {
+    let url;
+    const orderId = this.getOrderId();
+    if (orderId) {
+      // if any order has order copy to be uploaded
+      url = applyParms(OTHER_PROCEEDINGS_DOCUMENT_UPLOAD, { orderType: this.orderType, orderId });
+    } else {
+      // none of the orders in the current order type have order copy to be uploaded
+      const nextOrderType = this.getNextOrderType();
+      if (nextOrderType) {
+        url = applyParms(PROCEEDINGS_ORDER_DETAILS, { orderType: nextOrderType });
+      } else {
+        // there is no other order type present
+        if (isAnyOrderWithOrderCopy(caseData?.otherProceedings?.order)) {
+          // check at last if there were any previous order types having at least an order with order copy
+          url = COURT_PROCEEDINGS_SUMMARY;
+        } else {
+          url = PROCEEDINGS_COURT_PROCEEDINGS;
+        }
+      }
+    }
+    return url;
+  }
+  private getNextUrlOtherProceedingDocument() {
```

Comment:
Has there been test cases covered for the flow. Please make sure the test case coverage for this file is more than 95%.

Reply:
Added few test cases and now the test coverage is 94.64 for lines on this file